kasa/tapo/tapodevice.py

```diff
@@ -34,7 +34,7 @@ def __init__(
 
     async def update(self, update_children: bool = True):
         """Update the device."""
-        if self.credentials is None or self.credentials.username is None:
+        if not self.credentials:
```

Comment:
As per above I think we should keep the check for `self.credentials is None` but maybe drop the `or self.credentials.username is None`

Reply:
Agreed, I'm going to change that.